python/dgl/graphbolt/dataset.py

```diff
@@ -48,3 +53,67 @@ def graph(self) -> object:
     def feature(self) -> FeatureStore:
         """Return the feature."""
         raise NotImplementedError
+
+
+class OnDiskDataFormatEnum(pydantic_yaml.YamlStrEnum):
+    """Enum of data format."""
+
+    TORCH = "torch"
+    NUMPY = "numpy"
+
+
+class OnDiskTVTSet(pydantic.BaseModel):
+    """Train-Validation-Test set."""
+
+    type_name: str
+    format: OnDiskDataFormatEnum
+    path: str
+
+
+class OnDiskMetaData(pydantic_yaml.YamlModel):
+    """Metadata specification in YAML.
+
+    As multiple node/edge types and multiple splits are supported, each TVT set
+    is a list of list of ``OnDiskTVTSet``.
+    """
+
+    train_set: Optional[List[List[OnDiskTVTSet]]]
+    validation_set: Optional[List[List[OnDiskTVTSet]]]
+    test_set: Optional[List[List[OnDiskTVTSet]]]
+
+
+class OnDiskDataset(Dataset):
+    """An on-disk dataset.
+
+    An on-disk dataset is a dataset which stores the data on disk. It is
+    usually used when the dataset is too large to fit into the memory.
```

Comment:
This comment is a little bit confusing, what do you mean by the dataset is too large to fit into the memory? Before we run the code, everything should be on disk, and one we start training, some of the data will be load to memory, and some will remain on disk.

Can you clarify based on our offline discussion?

Reply:
sure, the comment here should be refined